### ISSUE TITLE
Model Upgrade: Model Accuracy Improvements

### ISSUE THREAD

**Issue by tamanna-dhir** [SrishtiSonam/SecureMedAI#8]
Accuracy & Model Performance Issues:

1. Improve Random Forest Accuracy with Hyperparameter Tuning:-
Current Behavior of the model:
Random Forest uses default hyperparameters.
This can lead to underfitting or overfitting.

Use GridSearchCV or RandomizedSearchCV to find optimal parameters
Replace current model with tuned model for better accuracy and F1-score

**Comment by tamanna-dhir** [comment]
please assign the level of the issue @SrishtiSonam
